pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -3093,9 +3236,124 @@ func (r *ClickHouseReader) GetMetricResult(ctx context.Context, query string) ([
 		series := model.Series{Labels: attributes, Points: points}
 		seriesList = append(seriesList, &series)
 	}
+	// err = r.db.Exec(ctx, "DROP TEMPORARY TABLE IF EXISTS getSubTreeSpans"+hash)
+	// if err != nil {
+	// 	zap.S().Error("Error in dropping temporary table: ", err)
+	// 	return nil, err
+	// }
+
 	return seriesList, nil
 }
 
+func getSubTreeAlgorithm(payload []model.SearchSpanResponseItem, getSpansSubQueryDBResponses []model.GetSpansSubQueryDBResponse) (map[string]*model.SearchSpanResponseItem, error) {
+
+	var spans []*model.Span
+	for _, spanItem := range payload {
+		var parentID string
+		if len(spanItem.References) > 0 && spanItem.References[0].RefType == "CHILD_OF" {
+			parentID = spanItem.References[0].SpanId
+		}
+		span := &model.Span{
+			TimeUnixNano: spanItem.TimeUnixNano,
+			SpanID:       spanItem.SpanID,
+			TraceID:      spanItem.TraceID,
+			ServiceName:  spanItem.ServiceName,
+			Name:         spanItem.Name,
+			Kind:         spanItem.Kind,
+			DurationNano: spanItem.DurationNano,
+			TagMap:       spanItem.TagMap,
+			ParentID:     parentID,
+			Events:       spanItem.Events,
+			HasError:     spanItem.HasError,
+		}
+		spans = append(spans, span)
+	}
+
+	zap.S().Debug("Building Tree")
+	roots, err := buildSpanTrees(&spans)
+	if err != nil {
+		return nil, err
+	}
+	searchSpansResult := make(map[string]*model.SearchSpanResponseItem)
+	for i, getSpansSubQueryDBResponse := range getSpansSubQueryDBResponses {
+		targetSpan := &model.Span{}
+		zap.S().Debug("Building tree for span id: " + getSpansSubQueryDBResponse.SpanID + " " + strconv.Itoa(i) + " of " + strconv.Itoa(len(getSpansSubQueryDBResponses)))
+		for _, root := range roots {
+			targetSpan, err = breadthFirstSearch(root, getSpansSubQueryDBResponse.SpanID)
+			if targetSpan != nil {
+				break
+			}
+			if err != nil {
+				zap.S().Error("Error during BreadthFirstSearch(): ", err)
+				return nil, err
+			}
+		}
+		if targetSpan == nil {
+			return nil, nil
+		}
+
+		targetSpan.ParentID = ""
+		preParents := []*model.Span{targetSpan}
+		children := []*model.Span{}
+
+		for i := 0; len(preParents) != 0; i++ {
+			parents := []*model.Span{}
+			for _, parent := range preParents {
+				children = append(children, parent.Children...)
+				parents = append(parents, parent.Children...)
+			}
+			preParents = parents
+		}
+		resultSpansSet := make(map[*model.Span]struct{})
+		resultSpansSet[targetSpan] = struct{}{}
+
+		for _, child := range children {
+			resultSpansSet[child] = struct{}{}
+		}
+
+		resultSpans := []*model.Span{}
+		for i := range resultSpansSet {
```

Comment:
I just see `resultSpansSet` being created and used without any processing. Please explain the use of  `resultSpansSet`